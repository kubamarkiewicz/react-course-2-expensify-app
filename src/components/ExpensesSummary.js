import React from 'react';
import { connect } from 'react-redux';
import getVisibleExpenses from '../selectors/expenses';
import getExpensesTotal from '../selectors/expenses-total.js';
import numeral from 'numeral';

export const ExpensesSummary = ({ expensesCount, expensesTotal }) => (
	<div>
		Viewing {expensesCount} expenses totalling {numeral(expensesTotal / 100).format('$0.00')}
	</div>
);



const mapStateToProps = (state) => {
	const expenses = getVisibleExpenses(state.expenses, state.filters);
	return {
		expensesCount: expenses.length,
		expensesTotal: getExpensesTotal(expenses),
	}
}

export default connect(mapStateToProps)(ExpensesSummary); 