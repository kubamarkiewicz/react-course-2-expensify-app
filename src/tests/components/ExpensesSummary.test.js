import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary.js';
import expenses from '../fixtures/expenses.js';

test('should render ExpensesSummary', () => {
	const wrapper = shallow(<ExpensesSummary expensesCount={12} expensesTotal={300} />)
	expect(wrapper).toMatchSnapshot();
});