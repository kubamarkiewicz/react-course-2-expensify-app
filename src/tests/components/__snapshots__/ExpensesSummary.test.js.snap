// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary 1`] = `
<div>
  Viewing 
  12
   expenses totalling 
  $3.00
</div>
`;
